### ISSUE TITLE
Minor Improvement to :GoFmt Requested

### ISSUE THREAD

**Issue by nobozo** [fatih/vim-go#3360]
### What did you do? (required: The issue will be **closed** when not provided)
I setup vim 8.2.4386 on Fedora 35 Server x86_64 to run :GoFmt when I write out a Go file.
This works fine, except because I make a lot of mistakes, the :GoFmt  output display often doesn't
start at error #1. This means I have to scroll the :GoFmt up to the first error. I've attached a
screen dump to this report.

I only have to save a Go file.
### What did you expect to happen?
I'd like the error display to start with error #1
### What happened instead?
The error display starts with an error after #1.
#### vim-go version:
I'm not sure how to see this. It's the version that I get today 2/17/2022.

####  `vimrc` you used to reproduce:

<details><summary>vimrc</summary>

```vim

```
</details>

#### Vim version (first three lines from `:version`):

VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Feb 15 2022 00:00:00)
Included patches: 1-4386
Modified by <bugzilla@redhat.com>
#### Go version (`go version`):

go version go1.17.7 linux/amd64
#### Go environment
<details><summary><code>go env</code> Output:</summary><br><pre>

GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/jonf/.cache/go-build"
GOENV="/home/jonf/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/jonf/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/jonf/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.17.7"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/dev/null"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build198855228=/tmp/go-build -gno-record-gcc-switches"
</pre></details>

#### gopls version
<details><summary><code>gopls version</code> Output:</summary><br><pre>

golang.org/x/tools/gopls v0.7.5
    golang.org/x/tools/gopls@v0.7.5 h1:8Az52YwcFXTWPvrRomns1C0N+zlgTyyPKWvRazO9GG8=
</pre></details>
<img width="754" alt="vim" src="https://user-images.githubusercontent.com/3916830/154618201-73759b47-1a65-4d02-9d1d-7ac54f128efe.PNG">

**Comment by bhcleek** [comment]
What value are you using for `g:go_fmt_command`?

**Comment by telemachus** [comment]
Open vim and run `:echo g:go_fmt_command`. That should show the current value.

**Comment by nobozo** [comment]
This variable isn't defined. I've attached a screen dump of what I see right after trying to write out a Go file.
GoFmt is working fine, except for the irritation of not starting its window with the first error.

<img width="663" alt="fmt" src="https://user-images.githubusercontent.com/3916830/154748594-1691618c-a75f-4160-82d0-0bbff40a0377.PNG">
.

**Comment by bhcleek** [comment]
I'm having a hard time getting enough errors and of the right types to duplicate what you're seeing. Do you have a file I could use as a test case?

**Comment by nobozo** [comment]
On 2/18/2022 11:36 AM, Billie Cleek wrote:
> I'm having a hard time getting enough errors and of the right types to 
> duplicate what you're seeing. Do you have a file I could use as a test case?

That file I'm showing in the screen dump is part of a much
larger package. I'll try create a minimal file that shows
the problem.

Thanks for looking at this.

Jon

**Comment by bhcleek** [comment]
I've been able to duplicate this with `g:go_fmt_command` set to `gofmt`, but not with the default, `gopls` -- `gopls` seems to abbreviate the number of errors in the cases I've tested with. That's ok, though, the overall effect is the same.

I don't see a good way to handle this from within vim-go, though. In my testing, the current cursor location in the location list window seems to depend on where the cursor is in the code window; I've seen the location list window be as you want it to be, as it is in your examples, and somewhere in the middle.

When vim-go opens the location list window, it sets the height according to `g:go_list_height`, and that seems to play some role in the behavior you're seeing, though I don't see a way to tell vim whether to reduce from the bottom of the window or from the top 🤔

**Comment by bhcleek** [comment]
I took another look at this today, and I realized that it would be relatively easy to solve by including a call in the right place in the formatting handlers, but I'm a little reluctant to do so. I'm not sure if it's the right thing to do when formatting. I'm in the habit of saving often and sometimes my code above the cursor line is not yet in a state where I'd want to fix it. Jumping the cursor around in that case would be disruptive, and I suspect others would feel the same.

However, there's a simple solution for you when do want to jump to the first. Assuming you're using the default list settings for the formatting errors, a `:ll 1` will take you to the first error location.
